samples/nrf9160/nrf_cloud_agps/src/main.c

```diff
@@ -0,0 +1,462 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <zephyr.h>
+#include <stdio.h>
+#include <modem/lte_lc.h>
+#include <net/cloud.h>
+#include <net/socket.h>
+#include <dk_buttons_and_leds.h>
+#include <drivers/gps.h>
+#include <nrf_cloud_agps.h>
+#include <power/reboot.h>
+
+#include <logging/log.h>
+
+LOG_MODULE_REGISTER(nrf_cloud_agps_sample,
+		    CONFIG_NRF_CLOUD_AGPS_SAMPLE_LOG_LEVEL);
+
+static struct cloud_backend *cloud_backend;
+static struct device *gps_dev;
+static u64_t start_search_timestamp;
+static u64_t fix_timestamp;
+static struct k_delayed_work cloud_update_work;
+static struct k_delayed_work gps_start_work;
+static struct k_delayed_work reboot_work;
+
+static void gps_start_work_fn(struct k_work *work);
+
+static void cloud_update_work_fn(struct k_work *work)
```

Comment:
I wonder if those periodic updates should be a part of this sample. All in all we're exposing agps functionaly here. Up for discussion.

Reply:
Yeah, they're not really used anyway. It only sends messageupon button press. Converted to regular function, not work item.